src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -32,6 +33,7 @@ public class PolicyService {
   public static final String GROUP_CONSTRAINT_POLICY_NAME = "group-constraint";
   public static final String GROUP_CONSTRAINT_KEY_NAME = "group";
   public static final String PROTECTED_DATA_POLICY_NAME = "protected-data";
+  public static final TpsUpdateMode UPDATE_MODE = TpsUpdateMode.FAIL_ON_CONFLICT;
```

Comment:
From TPS updatePao Swagger documentation:

```
If the updateMode is FAIL_ON_CONFLICT, the change is applied if there are no conflicts.
If the updateMode is ENFORCE_CONFLICT, then conflicts during the recomputing of effective attributes are allowed and are propagated to any PAOs dependent on this PAO.
With great power comes great responsibility.
```

Let me know if we want to `ENFORCE_CONFLICT` when updating PAOs instead of fail!

Reply:
`FAIL_ON_CONFLICT` seems like a safer option here (it seems like `ENFORCE_CONFLICT` handles the merge conflicts automatically?). We shouldn't get a conflict error though right? My understanding is that this would only happen if we were attempting to add additional `group-constraint` policies to a PAO that already has them set, but we only allow users to set this once. And if we add `group-constraint` policies to a PAO that has a `protected-data` policy then that's not a conflict.

Reply:
I agree, it did seem safer so that was my preference too… I can try adding an auth domain to my test snapshot, the one that already has an auth domain, and see what the behavior is!

Reply:
Yep, we get a 409 as expected in that scenario:
```
{
  "message": "Snapshot 0a4da971-3b28-4a40-af60-1704216e6989 already has an auth domain set: [20231024_policy_test]",
  "errorDetail": []
}
```

And yes, my manual verification (in the PR description) shows that we can add a `group_constraint` policy to a PAO with a `protected-data` policy and it doesn't produce a conflict error!